src/HeatMap/components/HeatMapContainer.ts

```diff
@@ -39,13 +40,29 @@ export default class HeatMapContainer extends Component<HeatMapContainerProps, H
         this.resetSubscriptions(newProps.mxObject);
         if (!this.state.alertMessage) {
             this.fetchData(newProps.mxObject);
+            if (newProps.mxObject) {
+                this.setRefreshInterval(newProps.refreshInterval);
```

Comment:
move `newProps.mxObject` check in to `setRefreshInterval` see comments below

Reply:
Changed it